chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,323 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+
+        assert did_wallet.did_info.current_inner is not None
+        assert did_wallet.did_info.origin_coin is not None
+
+        # Ensure we have an mint_total value
+        if mint_total is None:
+            mint_total = len(metadata_list)
+        assert isinstance(mint_number_start, int)
+        assert len(metadata_list) <= mint_total + 1 - mint_number_start
+
+        # Ensure we have a did coin and its next inner puzzle hash
+        if did_coin is None:
+            coins = await did_wallet.select_coins(uint64(1))
+            assert coins is not None
+            did_coin = coins.pop()
+        innerpuz: Program = did_wallet.did_info.current_inner
+        if new_innerpuzhash is None:
+            new_innerpuzhash = innerpuz.get_tree_hash()
+        # make the primaries for the DID spend
+        primaries = [
+            AmountWithPuzzlehash(
+                {"puzzlehash": new_innerpuzhash, "amount": uint64(did_coin.amount), "memos": [new_innerpuzhash]}
+            )
+        ]
+
+        # Ensure we have an xch coin of high enough amount
+        assert isinstance(fee, uint64)
+        total_amount = len(metadata_list) + fee
+        if xch_coins is None:
+            xch_coins = await self.standard_wallet.select_coins(uint64(total_amount))
+        assert len(xch_coins) > 0
+
+        # set the chunk size for the spend bundle we're going to create
+        chunk_size = len(metadata_list)
+
+        # Because bulk minting may not mint all the NFTs in one bundle, we
+        # calculate the edition numbers that will be used in the intermediate
+        # puzzle based on the starting edition number given, and the size of the
+        # chunk going into this spend bundle
+        mint_number_end = mint_number_start + chunk_size
+
+        # Empty set to load with the announcements we will assert from DID to
+        # match the announcements from the intermediate launcher puzzle
+        did_announcements: Set = set()
+        puzzle_assertions: Set = set()
+        amount = uint64(1)
+        intermediate_coin_spends = []
+        launcher_spends = []
+        launcher_ids = []
+        eve_spends = []
+        transfer_spends = []
+        p2_inner_puzzle = await self.standard_wallet.get_new_puzzle()
+
+        # Loop to create each intermediate coin, launcher, eve and (optional) transfer spends
+        for mint_number in range(mint_number_start, mint_number_end):
+            # Create  the puzzle, solution and coin spend for the intermediate launcher
+            intermediate_launcher_puz = did_wallet_puzzles.DID_NFT_LAUNCHER_MOD.curry(
+                did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, mint_number, mint_total
+            )
+            primaries.append(
+                AmountWithPuzzlehash(
+                    {
+                        "puzzlehash": intermediate_launcher_puz.get_tree_hash(),
+                        "amount": uint64(0),
+                        "memos": [intermediate_launcher_puz],
```

Comment:
You forgot `.get_tree_hash()` here but I also don't know why you would need this memo sine you're spending this right now anyways.  Probably a good idea to assign an `intermediate_launcher_puz_hash` before calling `get_tree_hash` a bunch of times.